atlasdb-commons/src/main/java/com/palantir/util/MathUtils.java

```diff
@@ -244,4 +246,20 @@ public static double keepSignificantFigures(double input, int sigfigs) {
     }
 
     private MathUtils() {/**/}
+
+    public static double confidenceDistributionIsNotUniform(List<Long> values) {
+        return 1.0 - confidenceDistributionIsUniform(values);
+    }
+
+    public static double confidenceDistributionIsUniform(List<Long> values) {
+        return Gamma.regularizedGammaQ((values.size() - 1.0) / 2,
+                chiSquareDistance(values) / 2);
+    }
+
+    private static double chiSquareDistance(List<Long> values) {
+        double mean = values.stream().mapToDouble(Long::doubleValue).reduce(0.0, Double::sum) / values.size();
```

Comment:
Probably more performant to sum the values as longs and then convert to a double just once at the end.